model/common/src/icon4py/model/common/interpolation/interpolation_fields.py

```diff
@@ -647,6 +674,31 @@ def compute_mass_conserving_bilinear_cell_average_weight(
     )
 
 
+def compute_mass_conserving_bilinear_cell_average_weight_torus(
+    c2e2c0: data_alloc.NDArray,
+    cell_areas: data_alloc.NDArray,
+    cell_owner_mask: data_alloc.NDArray,
+    divavg_cntrwgt: ta.wpfloat,
+    horizontal_start: gtx.int32,
+    horizontal_start_level_3: gtx.int32,
+    array_ns: ModuleType = np,
+) -> data_alloc.NDArray:
+    c_bln_avg = _compute_uniform_c_bln_avg(
+        c2e2c0[:, 1:], divavg_cntrwgt, horizontal_start, array_ns
+    )
+    # TODO(msimberg): Exact result for torus without the following. 1e-16 error
+    # with the the following. Is it needed?
+    return _force_mass_conservation_to_c_bln_avg(
+        c2e2c0,
+        c_bln_avg,
+        cell_areas,
+        cell_owner_mask,
+        divavg_cntrwgt,
+        horizontal_start_level_3,
+        array_ns,
+    )
```

Comment:
I can't really judge if there are cases for torus where `force_mass_conservation` is actually required for some grids.

If not, we could leave it out, but the inaccuracy that it introduces is also minimal, so may be safer to leave it in.